agent/src/agentclient.rs

```diff
@@ -92,8 +89,12 @@ impl<T: APIServerClient> BrupopAgent<T> {
         if !associated_bottlerocketshadow.is_empty() {
             Ok(true)
         } else {
+            // Build an incluster config used to infer the current namespace
+            let incluster_config =
+                kube::Config::incluster_dns().context(agentclient_error::K8sConfigCreateSnafu)?;
```

Comment:
We do this elsewhere where we explicitly create an incluster DNS client config:
https://github.com/bottlerocket-os/bottlerocket-update-operator/blob/f69a1708c3fb095b160665ca162fe4d2f602d6bc/apiserver/src/main.rs#L48-L49

And this is a workaround until we can get the next version of `kube-rs` with the `default_namespace()` function:
https://github.com/kube-rs/kube/pull/1123

Reply:
Thinking alittle more on this: we'll still need to get an incluster config so I wonder if a subsequent PR (once `default_namespace()` is available in `kube-rs`) would be to refactor these calls into a common shared library

Reply:
I would prefer if we clean this up now given how many times we repeat this pattern. It would make the move to `default_namespace()` a smaller change as well.

Reply:
Sounds good!! 😄